### PR TITLE
[CPDEV-108608] Close_all_high_and_critical_vulnerabilities_Kubemarine

### PR DESCRIPTION
**Pull request by nikhil1697**
### Description
Old k8s version v1.26 is removed
New calico plugin version v3.29.1 should be adopted
* 

### Test Cases
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

**TestCase 1**

Steps:

1. Kubemarine install

Results:

| Before | After |
| ------ | ------ |
| Not applicable | Success |

2. Kubemarine upgrade

Results:

| Before | After |
| ------ | ------ |
| Not applicable | Success |

### Checklist
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] There is no breaking changes, or migration patch is provided
- [ ] Integration CI passed
- [ ] Unit tests. If Yes list of new/changed tests with brief description
- [ ] There is no merge conflicts

#### Unit tests
Indicate new or changed unit tests and what they do, if any.

### CODE REVIEW

**Review by koryaga — APPROVED** [review]

File: kubemarine/resources/configurations/compatibility/kubernetes_versions.yaml [hunk 1]
```diff
@@ -81,61 +55,61 @@ compatibility_map:
...
   v1.29.7:
-    calico: v3.28.1
+    calico: v3.29.1
     nginx-ingress-controller: v1.11.1
     kubernetes-dashboard: v2.7.0
     local-path-provisioner: v0.0.27
     crictl: v1.30.0
   v1.30.1:
-    calico: v3.27.3
+    calico: v3.27.4
```
Comment (theboringstuff) [inline kubemarine/resources/configurations/compatibility/kubernetes_versions.yaml:100]: Why `v3.27.4` and not `v3.29.1`?

File: kubemarine/resources/schemas/definitions/services/kubeadm.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
   "properties": {
     "kubernetesVersion": {
       "type":  "string",
-      "default": "v1.26.11",
+      "default": "v1.27.13",
```
Comment (theboringstuff) [inline kubemarine/resources/schemas/definitions/services/kubeadm.json:8]: according to `defaults.yaml`, we use `v1.28.12` as a default, so maybe we should use it in json schema too?

Comment (theboringstuff) [reply kubemarine/resources/schemas/definitions/services/kubeadm.json:8]: Discussed with Sergey. It probably makes in scope of this activity to also change default version from 1.28.12 to 1.30.3 (current recommended).

File: kubemarine/resources/configurations/compatibility/kubernetes_versions.yaml [hunk 3]
```diff
@@ -81,61 +55,61 @@ compatibility_map:
...
   v1.28.9:
-    calico: v3.27.3
+    calico: v3.27.4
     nginx-ingress-controller: v1.10.1
     kubernetes-dashboard: v2.7.0
     local-path-provisioner: v0.0.26
     crictl: v1.30.0
   v1.28.12:
-    calico: v3.28.1
+    calico: v3.29.1
```
Comment (theboringstuff) [inline kubemarine/resources/configurations/compatibility/kubernetes_versions.yaml:76]: According to https://docs.tigera.io/calico/latest/getting-started/kubernetes/requirements#kubernetes-requirements

Calico 3.29 is not tested on 1.28, so probably better not change it

Comment (nikhil1697) [reply kubemarine/resources/configurations/compatibility/kubernetes_versions.yaml]: Okay kept v3.28.1 itself for v1.28.12

**Pull request merged**